search/ui/class-vip-search-widget.php

```diff
@@ -0,0 +1,454 @@
+<?php
+
+namespace Automattic\VIP\Search\UI;
+
+/**
+* Class VIP_Search_Widget
+ * @package Automattic\VIP\Search\UI
+ *
+ * Derived from Jetpack_Search_Widget
+ */
+class VIP_Search_Widget extends \WP_Widget {
+
+	/**
+	 * Default sort order for search results.
+	 *
+	 * @var string
+	 */
+	const DEFAULT_SORT = 'relevance_desc';
+
+	public function __construct( $name = null ) {
+		if ( empty( $name ) ) {
+			$name = 'Search (VIP Enterprise)';
+		}
+
+		parent::__construct(
+			'vip-search-widget',
+			$name,
+			array( 'description' => 'UI for VIP Enterprise Search' )
+		);
+
+		if ( ! is_admin() ) {
+			add_action( 'wp_enqueue_scripts', array( $this, 'enqueue_frontend_scripts' ) );
+		}
+	}
+
+	/**
+	 * Enqueue scripts and styles for the frontend.
+	 *
+	 */
+	public function enqueue_frontend_scripts() {
+		if ( ! is_active_widget( false, false, $this->id_base, true ) ) {
+			return;
+		}
+
+		wp_enqueue_script(
+			'vip-search-widget',
+			plugins_url( 'js/search-widget.js', __FILE__ ),
+			array(),
+			1.0,
+			true
+		);
+
+		wp_enqueue_style( 'vip-search-widget', plugins_url( 'css/search-widget-frontend.css', __FILE__ ), array(), 1.0 );
+	}
+
+	/**
+	 * Get the list of valid sort types/orders.
+	 *
+	 * @return array The sort orders.
+	 *
+	 */
+	private function get_sort_types():array {
+		return array(
+			'relevance|DESC' => is_admin() ? esc_html__( 'Relevance (recommended)', 'jetpack' ) : esc_html__( 'Relevance', 'jetpack' ),
+			'date|DESC'      => esc_html__( 'Newest first', 'jetpack' ),
+			'date|ASC'       => esc_html__( 'Oldest first', 'jetpack' ),
+		);
+	}
+
+	public function vip_search_populate_defaults( $instance ) {
+		$instance = wp_parse_args(
+			(array) $instance, array(
+				'title'              => '',
+				'search_box_enabled' => true,
+				'user_sort_enabled'  => true,
+				'sort'               => self::DEFAULT_SORT,
+				'post_types'         => array(),
+			)
+		);
+
+		return $instance;
+	}
+
+	/**
+	 * Responsible for rendering the widget on the frontend.
+	 *
+	 * @param array $args     Widgets args supplied by the theme.
+	 * @param array $instance The current widget instance.
+	 *
+	 */
+	public function widget( $args, $instance ) {
+		$instance = $this->vip_search_populate_defaults( $instance );
+
+		if ( empty( $instance['search_box_enabled'] ) && empty( $instance['user_sort_enabled'] ) ) {
+			return;
+		}
+
+		$title = ! empty( $instance['title'] ) ? $instance['title'] : '';
+
+		/** This filter is documented in core/src/wp-includes/default-widgets.php */
+		$title = apply_filters( 'widget_title', $title, $instance, $this->id_base );
```

Comment:
What would be the use-case of utilizing this core filter?

Reply:
This comes from Jetpack, I guess it's to provide additional flexibility. I'm not sure how I'd use it, so I'm happy to remove it, however, I don't think it hurts.